pgjdbc/src/test/java/org/postgresql/test/core/v3/V3ParameterListTests.java

```diff
@@ -0,0 +1,227 @@
+
+  public void testMergeOfParameterLists() {
+    try {
+      SocketFactory socketFactory = SocketFactoryFactory.getSocketFactory(System.getProperties());
+      HostSpec hostSpec = new HostSpec(TestUtil.getServer(), 5432);
+      PGStream stream = new PGStream(socketFactory, hostSpec);
+      ClassLoader cl = this.getClass().getClassLoader();
+      Class clsProtoConnImpl = Class.forName(
+          "org.postgresql.core.v3.ProtocolConnectionImpl", true, cl);
+      Constructor c = clsProtoConnImpl.getDeclaredConstructor(PGStream.class,
+          String.class, String.class, Properties.class, Logger.class,
+          Integer.TYPE);
+      c.setAccessible(true);
+      Object oPCI = c.newInstance(new Object[] { stream, "", "",
+          new Properties(), new Logger(), 5000 });
+      c = null;
+      assertNotNull(oPCI);
+
+      Class clsSPL = Class.forName(
+          "org.postgresql.core.v3.SimpleParameterList", true, cl);
+      Constructor initSPL = clsSPL.getDeclaredConstructor(new Class[] {
+          Integer.TYPE, clsProtoConnImpl });
+      initSPL.setAccessible(true);
+      Object s1SPL = initSPL.newInstance(8, oPCI);
+      assertNotNull(s1SPL);
+      Method msetIP = clsSPL.getMethod("setIntParameter", Integer.TYPE,
+          Integer.TYPE);
+      assertNotNull(msetIP);
+      msetIP.setAccessible(true);
+      msetIP.invoke(s1SPL, 1, 1);
+      msetIP.invoke(s1SPL, 2, 2);
+      msetIP.invoke(s1SPL, 3, 3);
+      msetIP.invoke(s1SPL, 4, 4);
+
+      Object s2SPL = initSPL.newInstance(new Object[] { 4, oPCI });
+      assertNotNull(s2SPL);
+      msetIP.invoke(s2SPL, 1, 5);
+      msetIP.invoke(s2SPL, 2, 6);
+      msetIP.invoke(s2SPL, 3, 7);
+      msetIP.invoke(s2SPL, 4, 8);
+
+      Method mappAll = clsSPL.getMethod("appendAll", ParameterList.class);
+      mappAll.setAccessible(true);
+      mappAll.invoke(s1SPL, s2SPL);
+
+      Method mgetValues = clsSPL.getMethod("getValues");
+      mgetValues.setAccessible(true);
+      Object oValues = mgetValues.invoke(s1SPL);
+      assertNotNull(oValues);
+      assertTrue(oValues instanceof Object[]);
+      Object[] arrValues = (Object[]) oValues;
+      assertNotNull(arrValues[0]);
+      assertNotNull(arrValues[1]);
+      assertNotNull(arrValues[2]);
+      assertNotNull(arrValues[3]);
+      assertNotNull(arrValues[4]);
+      assertNotNull(arrValues[5]);
+      assertNotNull(arrValues[6]);
+      assertNotNull(arrValues[7]);
+      assertTrue(arrValues[0] instanceof byte[]);
+      assertTrue(arrValues[1] instanceof byte[]);
+      assertTrue(arrValues[2] instanceof byte[]);
+      assertTrue(arrValues[3] instanceof byte[]);
+      assertTrue(arrValues[4] instanceof byte[]);
+      assertTrue(arrValues[5] instanceof byte[]);
+      assertTrue(arrValues[6] instanceof byte[]);
+      assertTrue(arrValues[7] instanceof byte[]);
+      byte[] b = new byte[4];
+      ByteConverter.int4(b, 0, 1);
+      assertTrue(Arrays.equals(b, (byte[]) arrValues[0]));
```

Comment:
I would prefer single toString and single `Assert.assertEquals(message, expected, actual)` call.

Current approach does not validate array length.